cmd/outline-ss-server/metrics.go

```diff
@@ -325,50 +514,37 @@ func newPrometheusOutlineMetrics(ip2info ipinfo.IPInfoMap) *outlineMetricsCollec
 			Name: "ports",
 			Help: "Count of open Shadowsocks ports",
 		}),
-		dataBytes: prometheus.NewCounterVec(
-			prometheus.CounterOpts{
-				Name: "data_bytes",
-				Help: "Bytes transferred by the proxy, per access key",
-			}, []string{"dir", "proto", "access_key"}),
-		dataBytesPerLocation: prometheus.NewCounterVec(
-			prometheus.CounterOpts{
-				Name: "data_bytes_per_location",
-				Help: "Bytes transferred by the proxy, per location",
-			}, []string{"dir", "proto", "location", "asn"}),
-		timeToCipherMs: prometheus.NewHistogramVec(
-			prometheus.HistogramOpts{
-				Name:    "time_to_cipher_ms",
-				Help:    "Time needed to find the cipher",
-				Buckets: []float64{0.1, 1, 10, 100, 1000},
-			}, []string{"proto", "found_key"}),
-	}
-}
-
-func (m *outlineMetricsCollector) collectors() []prometheus.Collector {
-	return []prometheus.Collector{
-		m.tcpCollector,
-		m.udpCollector,
-		m.tunnelTimeCollector,
-
-		m.buildInfo,
-		m.accessKeys,
-		m.ports,
-		m.dataBytes,
-		m.dataBytesPerLocation,
-		m.timeToCipherMs,
-	}
+	}, nil
 }
 
 func (m *outlineMetricsCollector) Describe(ch chan<- *prometheus.Desc) {
-	for _, collector := range m.collectors() {
-		collector.Describe(ch)
-	}
+	m.tcpCollector.Describe(ch)
+	m.udpCollector.Describe(ch)
+	m.tunnelTimeCollector.Describe(ch)
+	m.buildInfo.Describe(ch)
+	m.accessKeys.Describe(ch)
+	m.ports.Describe(ch)
 }
 
 func (m *outlineMetricsCollector) Collect(ch chan<- prometheus.Metric) {
-	for _, collector := range m.collectors() {
-		collector.Collect(ch)
+	m.tcpCollector.Collect(ch)
+	m.udpCollector.Collect(ch)
+	m.tunnelTimeCollector.Collect(ch)
+	m.buildInfo.Collect(ch)
+	m.accessKeys.Collect(ch)
+	m.ports.Collect(ch)
+}
+
+func (m *outlineMetricsCollector) getIPInfoFromAddr(addr net.Addr) ipinfo.IPInfo {
+	ipInfo, err := ipinfo.GetIPInfoFromAddr(m.ip2info, addr)
+	if err != nil {
+		slog.LogAttrs(nil, slog.LevelWarn, "Failed client info lookup.", slog.Any("err", err))
```

Comment:
I think there are many IPs that don't map, so I wonder if this raises to warning level. Will this pollute the logs?

Reply:
This is the level we log at today; I'm merely moving this out of `tcp.go` and `udp.go`, so I don't see an urgent need to change the level here. How common do you reckon it is? I haven't seen any reports of this causing issues, but I'm open to changing the level to `INFO`?